### PR TITLE
Use local GPG key for tini binary validation

### PR DESCRIPTION
**Pull request by jbertran**
Fixes: GH-770

This frees us from the flakiness of the distant keyserver.

Test with `docker build images/salt-master` in project root.

### CODE REVIEW

**Review by slaperche-scality — COMMENTED** [review]
You should explain *why* we do that in the commit message.

You did it in the PR description (but PR description and GitHub issues are useless when working offline), so you could simply copy your "This frees us from the flakiness of the distant keyserver." in the commit msg.

File: images/salt-master/Dockerfile [hunk 1]
```diff
@@ -10,8 +10,8 @@ ARG TINI_VERSION=v0.18.0
 ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
 ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini.asc /tini.asc
 
-RUN gpg --batch --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 595E85A6B1B4779EA4DAAEC70B588DFF0527A9B7 \
- && gpg --batch --verify /tini.asc /tini
+COPY source.key source.key
```
Comment (sayf-eddine-scality) [inline images/salt-master/Dockerfile:13]: shouldnt the image make target depends on this file too ? ;)

**Review by sayf-eddine-scality — CHANGES_REQUESTED** [review]

File: images/salt-master/Dockerfile (hunk 1 above)
Comment (jbertran) [reply images/salt-master/Dockerfile:13]: Well... it doesn't _have_ to.

I guess it makes sense to have a more sensible error message than a gpg error message when building if the repo ever gets to a state where the key is removed, but on the other hand we have no way to "build" it anyway so there's little point to having it function as a dependency.

For comprehensiveness though, I'll add it.

Comment (sayf-eddine-scality) [reply images/salt-master/Dockerfile:13]: I meant in case you change the key file the container img wont get rebuilt

**Review by sayf-eddine-scality — DISMISSED** [review]

**Review by slaperche-scality — APPROVED** [review]

**Comment by jbertran** [comment]
@bert-e approve

**Review by gdemonet — APPROVED** [review]
LGTM, I would add a minor suggestion: `source.key` is not really explicit for a filename, maybe something like `tini_gpg.key`?

**Comment by gdemonet** [comment]
Once again, I'm late to the party

**Pull request merged**